catalog/core/views.py

```diff
@@ -192,6 +192,7 @@ def get_context_data(self, number_of_publications=25, **kwargs):
         context['recently_updated'] = recently_updated
         list = [CacheNames.CONTRIBUTION_DATA.value + str(pub.id) for pub in recently_updated]
```

Comment:
Don't shadow Python builtins, especially core builtins like `list`.

Can you explain the reason you are prefixing the publication id with CacheNames.CONTRIBUTION_DATA.value?

Reply:
We are using various default cachename as prefix inorder for better understanding and common globals name. for e.g. 'citation:contribution_data:pub_id-'. We have listed those names under citation.graphviz.globals.py. So inorder to access(get) contribution data for specific publication from the cache values we need to add those prefix values and append the specific publication id.

Reply:
Thanks for the explanation. So I think there should be a larger refactoring that has to go in here.

1. Use the dictionary as it was intended, with an actual key lookup - iterating across the keys and values of a dictionary comparing the dict key with some generated string (each time!) is way kludgy and inefficient
2. add a property like `pub.contribution_cache_key` which returns `CacheNames.CONTRIBUTION_DATA.value + self.id` and get rid of the over-and-over `add_str`